pymatgen/io/aims/inputs.py

```diff
@@ -0,0 +1,609 @@
+    "total_density_integrable",
+    "long_range_potential",
+    "hartree_potential",
+    "xc_potential",
+    "delta_v",
+    "ion_dens",
+    "dielec_func",
+    "elf",
+)
+
+ALLOWED_AIMS_CUBE_TYPES_STATE = (
+    "first_order_density",
+    "eigenstate",
+    "eigenstate_imag",
+    "eigenstate_density",
+)
+
+ALLOWED_AIMS_CUBE_FORMATS = (
+    "cube",
+    "gOpenMol",
+    "xsf",
+)
+
+
+@dataclass
+class AimsCube(MSONable):
+    """Class representing the FHI-aims cubes
+
+    Attributes:
+        type (str): The value to be outputted as a cube file
+        origin (Sequence[float] or tuple[float, float, float]): The origin of the cube
+        edges (Sequence[Sequence[float]]): Specifies the edges of a cube: dx, dy, dz
+            dx (float): The length of the step in the x direction
+            dy (float): The length of the step in the y direction
+            dx (float): The length of the step in the x direction
+        points (Sequence[int] or tuple[int, int, int]): The number of points
+            along each edge
+        spinstate (int): The spin-channel to use either 1 or 2
+        kpoint (int): The k-point to use (the index of the list printed from
+            `output k_point_list`)
+        filename (str): The filename to use
+        format (str): The format to output the cube file in: cube, gOpenMol, or xsf
+        elf_type (int): The type of electron localization function to use (
+            see FHI-aims manual)
+    """
+
+    type: str = field(default_factory=str)
+    origin: Sequence[float] | tuple[float, float, float] = field(default_factory=lambda: [0.0, 0.0, 0.0])
+    edges: Sequence[Sequence[float]] = field(default_factory=lambda: 0.1 * np.eye(3))
+    points: Sequence[int] | tuple[int, int, int] = field(default_factory=lambda: [0, 0, 0])
+    format: str = "cube"
+    spinstate: int | None = None
+    kpoint: int | None = None
+    filename: str | None = None
+    elf_type: int | None = None
+
+    def __post_init__(self) -> None:
+        """Check the inputted variables to make sure they are correct
+
+        Raises:
+            ValueError: If any of the inputs is invalid
+        """
+        split_type = self.type.split()
+        if split_type[0] in ALLOWED_AIMS_CUBE_TYPES:
+            if len(split_type) > 1:
+                msg = f"Cube of type {split_type[0]} can not have a state associated with it"
+                raise ValueError(msg)
+        elif split_type[0] in ALLOWED_AIMS_CUBE_TYPES_STATE:
+            if len(split_type) != 2:
+                msg = f"Cube of type {split_type[0]} must have a state associated with it"
+                raise ValueError(msg)
+        else:
+            raise ValueError("Cube type undefined")
+
+        if self.format not in ALLOWED_AIMS_CUBE_FORMATS:
+            raise ValueError(f"{self.format} is invalid. Cube files must have a format of {ALLOWED_AIMS_CUBE_FORMATS}")
+
+        if self.spinstate is not None and (self.spinstate not in [1, 2]):
```

Comment:
```suggestion
        if self.spinstate not in (None, 1, 2):
```

Reply:
and change err msg to clarify `None` is valid too

Reply:
good point